actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/BatchOperationNotifierActor.java

```diff
@@ -0,0 +1,421 @@
+              (String) requestMap.get(JsonKey.OPERATION_TYPE));
+          break;
+        }
+      default:
+        onReceiveUnsupportedOperation(request.getOperation());
+        break;
+    }
+  }
+
+  /*
+   * This method process invite-only batch enrol and unenroll operation
+   * notifications mentor and learner add remove Operation
+   *
+   * @param CourseBatch object for getting course and email related data
+   *
+   * @param operationType String for specifying the operation type ie: add/remove
+   */
+  private void batchCreateOperationNotifier(CourseBatch courseBatch, String operationType) {
+    if (operationType.equals(JsonKey.ADD)) {
+      sendEmailNotification(courseBatch, JsonKey.ADD);
+    } else {
+      sendEmailNotification(courseBatch, JsonKey.REMOVE);
+    }
+  }
+
+  /*
+   * This method process open batch enroll and unenroll operation notifications
+   *
+   * @param courseMap , map for getting operation and email related data
+   *
+   * @param courseBatch object for getting course and email related data
+   */
+  @SuppressWarnings("unchecked")
+  public void batchEnrollOperationNotifier(
+      Map<String, Object> courseMap, CourseBatch courseBatch, String operationType) {
+    Map<String, Object> requestMap = createRequestMap(courseBatch);
+    List<String> userId = new ArrayList<>();
+    userId.add((String) courseMap.get(JsonKey.USER_ID));
+    List<Map<String, Object>> participentList = getUsersFromDB(userId);
+    Map<String, String> user;
+    if (!CollectionUtils.isEmpty(participentList)) {
+      if (operationType.equals(JsonKey.ADD)) {
+        user = getUserData(participentList.get(0), JsonKey.ADD);
+        user.put(JsonKey.TEMPLATE_NAME, JsonKey.BATCH_LEARNER_ENROL);
+      } else {
+        user = getUserData(participentList.get(0), JsonKey.REMOVE);
+        user.put(JsonKey.TEMPLATE_NAME, JsonKey.OPEN_BATCH_LEARNER_UNENROL);
+      }
+      sendMail(user, requestMap);
+    }
+  }
+
+  /*
+   * This method takes courseBatch object and ObjectType as parameter and process
+   * update batch related operation notifications
+   *
+   * @param CourseBatch object for course related data
+   *
+   * @param operationType String for specifying the operation type ie: add/remove
+   */
+  @SuppressWarnings("unchecked")
+  private void sendEmailNotification(CourseBatch courseBatch, String operationType) {
+    List<Map<String, Object>> mentorList = null;
+    List<Map<String, Object>> participentList = null;
+    mentorList = getUsersFromDB(courseBatch.getMentors());
+    List<String> userIds = getParticipants(courseBatch);
+    if (userIds != null) {
+      participentList = getUsersFromDB(userIds);
+    }
+    if (mentorList != null) {
+      processUserDataAndSendMail(mentorList, courseBatch, operationType, JsonKey.MENTOR);
+    }
+    if (participentList != null) {
+      processUserDataAndSendMail(mentorList, courseBatch, operationType, JsonKey.PARTICIPANT);
```

Comment:
mentorList is used in sending email for participants.

Reply:
noted. will change and commit .